src/main/java/updater/UpdateManager.java

```diff
@@ -0,0 +1,174 @@
+package updater;
+
+import javafx.application.Platform;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import ui.UpdateProgressWindow;
+import util.DialogMessage;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+
+/**
+ * UpdateManager is the class that will handle updating of HubTurbo application
+ */
+public class UpdateManager {
+    private static final Logger logger = LogManager.getLogger(UpdateManager.class.getName());
+
+    // Error messages
+    private static final String ERROR_INIT_UPDATE = "Failed to initialize update";
+    private static final String ERROR_DOWNLOAD_UPDATE_DATA = "Failed to download update data";
+    private static final String ERROR_DOWNLOAD_UPDATE_APP = "Failed to download updated application";
+
+    // Directories and file names
+    private static final String UPDATE_DIRECTORY = "updates";
+    private static final String UPDATE_SERVER_DATA_NAME =
+            "https://raw.githubusercontent.com/HubTurbo/AutoUpdater/master/HubTurbo.xml";
+    private static final String UPDATE_LOCAL_DATA_NAME = "HubTurbo.json";
+    private static final String UPDATE_APP_NAME = "HubTurbo.jar";
+
+    // Class member variables
+    private final UpdateProgressWindow updateProgressWindow;
+
+    private final ExecutorService pool = Executors.newSingleThreadExecutor();
+
+    public UpdateManager(UpdateProgressWindow updateProgressWindow) {
+        this.updateProgressWindow = updateProgressWindow;
+    }
+
+    /**
+     * Driver method to trigger UpdateManager to run. Update will be run on another thread.
+     *
+     * - Run is not automatic upon instancing the class in case there would like to be conditions on when to run update,
+     *   e.g. only if user is logged in
+     */
+    public void run() {
+        pool.execute(() -> runUpdate());
+    }
+
+    /**
+     * Method that will run update sequence.
+     * 1. Initialize system for update, e.g. make directory
+     * 2. Check if there is any new update
+     *   2.a. If there is no new update but the previously downloaded update is not applied yet,
+     *        prompt user for updating and restarting application
+     * 3. Download update according to user preference
+     * 4. Once update is done, prompt user for updating and restarting application
```

Comment:
Maybe we could check this inside `download` and throw an exception there instead?

Also, it would be useful if the three methods below this had comments describing the circumstances in which they fail.